### PR TITLE
Experimental Ingress Class Support

### PR DESCRIPTION
**Pull request by venezia**
There is a [ticket](https://github.com/jetstack/kube-lego/issues/233) that describes how to specify the ingress controller for kube-lego.  This adds that to the chart.

Also updates .gitlab-ci.yml and Jenkinsfile

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  stage: deploy
+  only:
+    - branches
+    - master
+  dependencies:
+  - build-branch
+  script:
+  - helm registry login -u ${REGISTRY_USERNAME} -p ${REGISTRY_PASSWORD} ${REGISTRY_HOST}
+  - cd ${CHART_NAME}
+  - helm registry push ${REGISTRY_CHART_URI} -c alpha
```
Comment (coffeepac) [inline .gitlab-ci.yml]: no beta channel?

Comment (venezia) [reply .gitlab-ci.yml]: Generally speaking, branch releases are "alpha" and tagged releases are "stable" .. unsure what could constitute beta.

Comment (coffeepac) [reply .gitlab-ci.yml]: I had that thought as well.  probably a well named branch 'beta' or something.  then feature branches get merged into beta and beta goes to master or something.  super not important, just curious.

File: kube-lego/values.yaml [hunk 2]
```diff
@@ -1,3 +1,6 @@
 image: jetstack/kube-lego:0.1.5
```
Comment (coffeepac) [inline kube-lego/values.yaml]: do you need to upgrade to a pre-release tag?  the linked issue and assorted PRs suggest that this won't work with 0.1.5, you need at least `0.1.6-dev-784fc8fe` or something

Comment (venezia) [reply kube-lego/values.yaml]: This is just to add support for it, it isn't defaulted on.  I would presume that you would use an image that supports it if you enable it.

Comment (coffeepac) [reply kube-lego/values.yaml]: okay.

**Pull request merged**